clouddriver-elasticsearch/src/main/java/com/netflix/spinnaker/clouddriver/elasticsearch/ElasticSearchServerGroupTagger.java

```diff
@@ -0,0 +1,101 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.elasticsearch;
+
+import com.netflix.spinnaker.clouddriver.core.services.Front50Service;
+import com.netflix.spinnaker.clouddriver.data.task.DefaultTask;
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository;
+import com.netflix.spinnaker.clouddriver.elasticsearch.descriptions.UpsertEntityTagsDescription;
+import com.netflix.spinnaker.clouddriver.elasticsearch.model.ElasticSearchEntityTagsProvider;
+import com.netflix.spinnaker.clouddriver.elasticsearch.ops.UpsertEntityTagsAtomicOperation;
+import com.netflix.spinnaker.clouddriver.model.EntityTags;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider;
+import com.netflix.spinnaker.clouddriver.tags.ServerGroupTagger;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ElasticSearchServerGroupTagger implements ServerGroupTagger {
+  private static final String SERVER_GROUP_TYPE = "servergroup";
+
+  private static final String ALERT_TYPE = "alert";
+  private static final String ALERT_KEY_PREFIX = "spinnaker_ui_alert:";
+
+  private final Front50Service front50Service;
+  private final AccountCredentialsProvider accountCredentialsProvider;
+  private final ElasticSearchEntityTagsProvider entityTagsProvider;
+
+  public ElasticSearchServerGroupTagger(Front50Service front50Service,
+                                        AccountCredentialsProvider accountCredentialsProvider,
+                                        ElasticSearchEntityTagsProvider entityTagsProvider) {
+    this.front50Service = front50Service;
+    this.accountCredentialsProvider = accountCredentialsProvider;
+    this.entityTagsProvider = entityTagsProvider;
+  }
+
+  @Override
+  public void alert(String cloudProvider,
+                    String accountId,
+                    String region,
+                    String serverGroupName,
+                    String key,
+                    String value) {
+    UpsertEntityTagsAtomicOperation upsertEntityTagsAtomicOperation = new UpsertEntityTagsAtomicOperation(
+      front50Service,
+      accountCredentialsProvider,
+      entityTagsProvider,
+      upsertEntityTagsDescription(cloudProvider, accountId, region, serverGroupName, key, value)
+    );
+
+    try {
+      TaskRepository.threadLocalTask.set(new DefaultTask(this.getClass().getSimpleName()));
+      upsertEntityTagsAtomicOperation.operate(Collections.emptyList());
+    } finally {
+      TaskRepository.threadLocalTask.set(null);
+    }
```

Comment:
Not a lock but rather a reference to the underlying persisted clouddriver task.

In this situation we're running an AtomicOperation outside the scope of a persisted task and this is a workaround to requiring an actual task.

The usage of said Task is ...

`getTask().updateStatus(BASE_PHASE, format("Tagged %s in Front50", durableEntityTags.getId()));`